src/Application/Actions/Hooks/StripeChargeUpdate.php

```diff
@@ -249,11 +291,48 @@ private function handleChargeRefunded(Event $event, Response $response): Respons
             return $this->respond($response, new ActionPayload(204));
         }
 
-        $this->doPostMarkRefundedUpdates($donation, $isFullRefund);
+        $this->doPostMarkRefundedUpdates($donation, $isFullRefund || $isOverRefund);
 
         return $this->respondWithData($response, $charge);
     }
 
+    /**
+     * @param string $eventType
+     * @param Donation $donation
+     * @param int $refundedOrDisputedAmount In small currency unit, e.g. pence or cents.
+     * @param string $refundedOrDisputedCurrencyCode
+     */
```

Comment:
IMHO easier to see the important parts if we leave out the docblock part that jsut repeats what's in the code:
```suggestion
     * @param int $refundedOrDisputedAmount In small currency unit, e.g. pence or cents.
     */
```

I appreciate the specification that $refundedOrDisputedAmount is in a small currency unit - but also the fact that we need to specify it makes me think it'd would be good to introduce a value object type and use that instead of `int`. I'd probably combine the amount with the currency So the calling code could look something like:

```php
            $this->warnAboutOverRefund(
                'charge.dispute.closed (lost)',
                $donation,
                Money::of(currencySymbol: $dispute->currency, minorUnits: $dispute->amount)
            );
```

Reply:
on the other hand as this is a private function it's not a big issue to manually check that the usage semantics are correct. Just a more general point.